pkg/registry/core/service/storage/storage_test.go

```diff
@@ -5832,4 +5834,979 @@ func TestDeleteDryRun(t *testing.T) {
+			ctx := genericapirequest.NewDefaultContext()
+
+			// Create the object as specified and check the results.
+			obj, err := storage.Create(ctx, tc.create.svc, rest.ValidateAllObjectFunc, &metav1.CreateOptions{})
+			if tc.create.expectError && err != nil {
+				return
+			}
+			if err != nil {
+				t.Fatalf("unexpected error creating service: %v", err)
+			}
+			defer storage.Delete(ctx, tc.create.svc.Name, rest.ValidateAllObjectFunc, &metav1.DeleteOptions{}) // in case
+			if tc.create.expectError && err == nil {
+				t.Fatalf("unexpected success creating service")
+			}
+			createdSvc := obj.(*api.Service)
+			if !verifyEquiv(t, "create", &tc.create, createdSvc) {
+				return
+			}
+			if tc.create.expectClusterIPs {
+				proveClusterIPsAllocated(t, storage, tc.create.svc, createdSvc)
+			} else if tc.create.expectHeadless {
+				proveHeadless(t, storage, tc.create.svc, createdSvc)
+			} else {
+				proveClusterIPsDeallocated(t, storage, tc.create.svc, createdSvc)
+			}
+			if tc.create.expectNodePorts {
+				proveNodePortsAllocated(t, storage, tc.create.svc, createdSvc)
+			} else {
+				proveNodePortsDeallocated(t, storage, tc.create.svc, createdSvc)
+			}
+			if tc.create.expectHealthCheckNodePort {
+				proveHealthCheckNodePortAllocated(t, storage, tc.create.svc, createdSvc)
+			} else {
+				proveHealthCheckNodePortDeallocated(t, storage, tc.create.svc, createdSvc)
+			}
+
+			// Update the object to the new state and check the results.
+			obj, created, err := storage.Update(ctx, tc.update.svc.Name,
+				rest.DefaultUpdatedObjectInfo(tc.update.svc), rest.ValidateAllObjectFunc,
+				rest.ValidateAllObjectUpdateFunc, false, &metav1.UpdateOptions{})
+			if tc.update.expectError && err != nil {
+				return
+			}
+			if err != nil {
+				t.Fatalf("unexpected error updating service: %v", err)
+			}
+			if tc.update.expectError && err == nil {
+				t.Fatalf("unexpected success updating service")
+			}
+			if created {
+				t.Fatalf("unexpected create-on-update")
+			}
+			updatedSvc := obj.(*api.Service)
+			if !verifyEquiv(t, "update", &tc.update, updatedSvc) {
+				return
+			}
+			if tc.update.expectClusterIPs {
+				proveClusterIPsAllocated(t, storage, createdSvc, updatedSvc)
+			} else if tc.update.expectHeadless {
+				proveHeadless(t, storage, createdSvc, updatedSvc)
+			} else {
+				proveClusterIPsDeallocated(t, storage, createdSvc, updatedSvc)
+			}
+			if tc.update.expectNodePorts {
+				proveNodePortsAllocated(t, storage, createdSvc, updatedSvc)
+			} else {
+				proveNodePortsDeallocated(t, storage, createdSvc, updatedSvc)
+			}
+			if tc.update.expectHealthCheckNodePort {
+				proveHealthCheckNodePortAllocated(t, storage, createdSvc, updatedSvc)
+			} else {
+				proveHealthCheckNodePortDeallocated(t, storage, createdSvc, updatedSvc)
+			}
+
+			// Delete the object and check the results.
+			_, _, err = storage.Delete(ctx, tc.update.svc.Name, rest.ValidateAllObjectFunc, &metav1.DeleteOptions{})
```

Comment:
Commit `Service REST: feature tests`: 
We are deferring a Delete() call L6043. And we are deferring ...Store.Destroy() -- Let us keep the destroy, remove the deferred delete. It just lengthen the tests duration.

Reply:
The defer'ed delete is to cover what happens if we abort the test early (`Fatalf()`).  The only way to avoid that need is to move the storage into the Run() func, and that makes the test REALLY slow when there are lots of cases